### PR TITLE
Feat/updated diamond selector

### PR DESCRIPTION
**Pull request by superzordon**
Todo:
 - Allow mobile to use regular click interface when selecting diamonds rather than just dragging
 - Refactor html template of selectable diamond elements to make it more dynamic, use some sort of loop
 - Enable question-mark on mobile dragging interface
 

https://user-images.githubusercontent.com/88362450/129152906-46552a14-233a-4ce6-8e18-c5f633a4eb4a.mov

https://user-images.githubusercontent.com/88362450/129152912-e0bff3a7-2c25-4236-948e-7b2f18d66fc6.mov

https://user-images.githubusercontent.com/88362450/129152916-846d6e17-8b29-4652-810e-e718c940af07.mov

### CODE REVIEW

**Comment by maebeam** [comment]
😍

**Comment by diamondhands0** [comment]
This looks incredible. A few high-level thoughts:

- I love drag up to select. Let's try that on a non-prod node and see how we feel about it.
- How come you don't show numbers until you hover over the particular diamond on desktop? I'm conflicted on whether it's better to just have them there. I think it probably is though?
- The help text needs to be updated since diamonds are in CLOUT now. See my copy that I pushed yesterday.
- The FE code itself looks a little kludge-ey but let's see what @lazynina thinks. If it achieves the desired effect I don't think it's a big deal.

File: src/app/feed/feed-post-icon-row/feed-post-icon-row.component.html [hunk 1]
```diff
@@ -83,87 +88,132 @@
...
+          'fas fa-spinner fa-spin': sendingDiamonds,
+          'fc-blue': postContent.PostEntryReaderState?.DiamondLevelBestowed > 0
+        }"
+        [ngStyle]="{
+          visibility: diamondDragging ? 'hidden' : 'visible'
+        }"
+      ></i>
+      <a
+        [ngClass]="{ 'dragged-like': diamondDragging }"
+        class="like-btn icon-diamond"
```
Comment (lazynina) [inline src/app/feed/feed-post-icon-row/feed-post-icon-row.component.html:153]: like-btn -- what's going on with this class?

File: src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts [hunk 2]
```diff
@@ -10,6 +10,8 @@ import { BsModalService } from "ngx-bootstrap/modal";
 import { CommentModalComponent } from "../../comment-modal/comment-modal.component";
 import { PopoverDirective } from "ngx-bootstrap/popover";
 import { ThemeService } from "../../theme/theme.service";
+import * as _ from "lodash";
```
Comment (lazynina) [inline src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts:13]:
```suggestion
```

**Review by lazynina — CHANGES_REQUESTED** [review]
See feedback - generally looks good, but could use some more comments on some of the new functions

File: src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts [hunk 3]
```diff
@@ -59,6 +79,73 @@ export class FeedPostIconRowComponent {
...
+
+  // Make sure that if a mobile device rotates, that the drag markers remain in the same place
+  @HostListener("window:orientationchange", ["$event"])
+  onOrientationChange() {
+    this.resetDragPosition();
+  }
+
+  resetDragPosition() {
+    setTimeout(() => {
+      const likeBtn = document.getElementById("diamond-select");
```
[the remark this answers is not in the document]
Comment (lazynina) [reply src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts:92]: also add some comments here on what's going on

File: src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts [hunk 4]
```diff
@@ -59,6 +77,73 @@ export class FeedPostIconRowComponent {
...
+      const leftOffset = this.getPosition(likeBtn).offsetLeft;
+      this.diamondDragLeftOffset = `${leftOffset}px`;
+    }, 200);
+  }
+
+  getPosition(element){
+    let offsetLeft = 0;
+    let offsetTop = 0;
+
+    while (element) {
```
[the remark this answers is not in the document]
Comment (lazynina) [reply src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts:102]: I think if you have an event, you may be able to get the offset without computing it like this.

File: src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts [hunk 5]
```diff
@@ -399,55 +452,81 @@ export class FeedPostIconRowComponent {
...
+      return;
+    }
+
+    // Block user from selecting diamond level below already gifted amount
+    if (this.getCurrentDiamondLevel() > 0) {
+      return;
+    }
+
+    // Don't trigger diamond purchases on tap on tablet
+    if (event && event.pointerType === "touch" && !fromDragEvent) {
```
Comment (lazynina) [inline src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts:474]: how do I send a single diamond on tablet / mobile?

File: src/styles.scss [hunk 6]
```diff
@@ -503,6 +503,7 @@ $green-font-color: #19B028;
 }
 
 .global__bottom-bar-mobile {
+  z-index: 2;
```
Comment (lazynina) [inline src/styles.scss:506]: do we want the z-index to be the same for the bottom bar and the create post circle?

File: src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (superzordon) [reply src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts:107]: We need the draggable div to be in position above the diamond button for the user to drag, so it needs to be in position on page load. Unfortunately, it sits inside an absolute-positioned div that spans the entire window, so there's no way for us to relatively-position it, thus the javascript hack-around to get there.

File: src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts (hunk 5 above)
Comment (superzordon) [reply src/app/feed/feed-post-icon-row/feed-post-icon-row.component.ts:485]: On tablet, you can tap the diamond to open the pop-up, and click again to select the amount. On mobile you tap to send a single diamond, or drag to select another amount.

File: src/styles.scss (hunk 6 above)
Comment (superzordon) [reply src/styles.scss:506]: Good point, updated create post circle

**Comment by maebeam** [comment]
UI Request: Can we get the same blue color on hover on mobile?

**Review by lazynina — APPROVED** [review]

**Pull request merged**